uint8[18] pin_config
bool[18] pin_value